tests/test_fastapi_extension.py

```diff
@@ -22,6 +22,50 @@ async def t1(request: Request):
             response = client.get("/t1")
             assert response.status_code == 200 if i < 5 else 429
 
+
+    def test_single_decorator_with_headers(self):
+        app, limiter = self.build_fastapi_app(key_func=get_ipaddr, headers_enabled=True)
+
+        @app.get("/t1")
+        @limiter.limit("5/minute")
+        async def t1(request: Request):
+            return PlainTextResponse("test")
+
+        client = TestClient(app)
+        for i in range(0, 10):
+            response = client.get("/t1")
+            assert response.status_code == 200 if i < 5 else 429
+            assert response.headers.get('X-RateLimit-Limit') is not None if i < 5 else True
+            assert response.headers.get('Retry-After') is not None if i < 5 else True
+
+    def test_single_decorator_not_response(self):
+        app, limiter = self.build_fastapi_app(key_func=get_ipaddr)
+
+        @app.get("/t1")
+        @limiter.limit("5/minute")
+        async def t1(request: Request, response: Response):
+            return {"key": "value"}
+
+        client = TestClient(app)
+        for i in range(0, 10):
+            response = client.get("/t1")
+            assert response.status_code == 200 if i < 5 else 429
+
+    def test_single_decorator_not_response_with_headers(self):
+        app, limiter = self.build_fastapi_app(key_func=get_ipaddr, headers_enabled=True)
+
+        @app.get("/t1")
+        @limiter.limit("5/minute")
+        async def t1(request: Request):
+            return PlainTextResponse("test")
```

Comment:
Shouldn't this `return {"key": "value"}` instead? I think this is the same as 2 tests above otherwise.

Reply:
ah, good spot! copy / paste and I forgot the response in the endpoint :(.
This should be fixed now.